### PR TITLE
Add test for algorithm provider and policy config file

### LINKED ISSUE #88028: Scheduler: Add tests for server.go for close-to-full E2E setup
We are lacking tests that exercise a full scheduler setup, all the way from config files and command line flags to what plugins end up instantiated.

The best candidate for this is `cmd/kube-scheduler/app/server.go`, but this needs tweaking. Basically, we need to split `runCommand` in two functions: `setupCommand` and the already existing  `Run` (which already has a test). Maybe we can have the test in `cmd/kube-scheduler/app/testing` as well.

Configuration can be written to files just like we do in `cmd/kube-scheduler/app/options/options_test.go` and args can be passed as parameters.

/sig scheduling
/help
/kind cleanup

### PR DESCRIPTION
**Pull request by notpad**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Add test for -algorithm-provider, --policy-config-file

**Which issue(s) this PR fixes**:

Fixes #88028

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kube-scheduler/app/server_test.go [hunk 1]
```diff
@@ -147,6 +147,58 @@ profiles:
 		t.Fatal(err)
 	}
 
+	// policy config file
+	policyConfigFile := filepath.Join(tmpDir, "policy-config.yaml")
+	if err := ioutil.WriteFile(policyConfigFile, []byte(`{
+		"kind": "Policy",
+		"apiVersion": "v1",
+		"predicates": [
+		  {"name": "MatchNodeSelector"},
```
Comment (alculquicondor) [inline cmd/kube-scheduler/app/server_test.go]: I would just make this minimal. This is not a compatibility test. I just want to make sure the flag works.

Comment (notpad) [reply cmd/kube-scheduler/app/server_test.go]: done

**Comment by alculquicondor** [comment]
/approve

please squash

**Comment by notpad** [comment]
squashed

**Comment by alculquicondor** [comment]
/lgtm

Thanks! :)

**Pull request merged**